src/components/CourseChecklist/index.jsx

```diff
@@ -329,8 +329,11 @@ class CourseChecklist extends React.Component {
 
   updateChecklistState(props) {
     if (Object.keys(props.data).length > 0) {
-      const checks = getFilteredChecklist(
-        props.dataList, props.data.is_self_paced);
+      const isSelfPaced = props.data.is_self_paced;
+      const hasCertificatesEnabled = props.data.certificates && props.data.certificates.is_enabled;
+      const hasHighlightsEnabled = props.data.sections && props.data.sections.highlights_enabled;
+      const checks = getFilteredChecklist(props.dataList,
+        isSelfPaced, hasCertificatesEnabled, hasHighlightsEnabled);
```

Comment:
This is much easier to follow (same as the OutlineStatus change) - thank you for the change!